### ISSUE TITLE
Question about result on DAVIS16

### ISSUE THREAD

**Issue by yongliu20** [hkchengrex/Mask-Propagation#33]
I want ask why I get this result using your pre-trained model?  Thanks!
![image](https://user-images.githubusercontent.com/72915904/120797989-4b484000-c56f-11eb-97da-38f012189039.png)

**Comment by hkchengrex** [comment]
That doesn't look right... Can I see some output images?

**Comment by hkchengrex** [comment]
Specifically, you can compare them with our pre-computed results.

**Comment by yongliu20** [comment]
The results of 'blackswan' are these:
![image](https://user-images.githubusercontent.com/72915904/120799854-d4f90d00-c571-11eb-8cb7-f72e8eca28ab.png)

**Comment by yongliu20** [comment]
I also think this J&F result is not right but I can not find out what went wrong...

**Comment by hkchengrex** [comment]
"blackswan" will almost always be good because it is so easy... Can you download our pre-computed results and see if there are any differences? I want to check whether the problem lies in generation or evaluation.

**Comment by yongliu20** [comment]
Ok, I will do it now

**Comment by yongliu20** [comment]
There are indeed some differences between them but very small. And I test your pre-computed results, the J&F is:
![image](https://user-images.githubusercontent.com/72915904/120805755-b77b7180-c578-11eb-9b03-18dc8f9db706.png)
The evaluation method which I use is from davis.

**Comment by yongliu20** [comment]
I find that my result is as same as the D16_s012. And the above result belongs to D16_s012_notop.

**Comment by hkchengrex** [comment]
1. I just evaluated again and I got the same reported number (90.9 for D16_s012_notop). To check that my evaluation code is correct, I downloaded STM's results (https://github.com/seoungwugoh/STM) (mapped from 0\~1 to 0\~255) and evaluated them in my environment. I got the same result as the reported number in STM. So most likely your evaluation has some problems.
2. Visually, the results also look closer to 90 than 75.
3. I also found that I swapped two rows (D16_s02, D16_s012_notop) in the table in README. I will fix that, thanks for the hint.

DAVIS 2016 evaluation code is not very well maintained. It was not easy to get it right for me back then... I recalled there are some discussion threads about a proper implementation. I am going to check on those and get back to you.

**Comment by hkchengrex** [comment]
Here: https://github.com/davisvideochallenge/davis2017-evaluation/issues/4
Hope it helps.

You can always check the numbers with ours/STM's.
I might open source my own evaluation code later.

**Comment by yongliu20** [comment]
Well, I have solved this problem by modifying your eval_davis_2016.py, like this:
![image](https://user-images.githubusercontent.com/72915904/120972331-38688200-c7a0-11eb-9cc9-e7afad9d1c95.png)

**Comment by hkchengrex** [comment]
Well yeah if your evaluation script expects 0/1 outputs... The ground truths in DAVIS 2016 are 0/255 so I'm sticking with that. Glad that it has been fixed.

**Comment by hkchengrex** [comment]
Hmm I think I can actually modify the code a bit to make both happy. Gonna do that. Thanks.

**Comment by yongliu20** [comment]
My previous problem is that the output values of the foreground pixels are not same. And thanks for your help!
